pkg/proxy/healthcheck/proxier_health.go

```diff
@@ -109,18 +120,44 @@ type healthzHandler struct {
 }
 
 func (h healthzHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	lastUpdated := time.Time{}
+	var lastQueued, lastUpdated time.Time
+	if val := h.hs.lastQueued.Load(); val != nil {
+		lastQueued = val.(time.Time)
+	}
 	if val := h.hs.lastUpdated.Load(); val != nil {
 		lastUpdated = val.(time.Time)
 	}
 	currentTime := h.hs.clock.Now()
 
+	healthy := false
+	switch {
+	case lastUpdated.IsZero():
+		// The proxy is healthy while it's starting up
+		// TODO: this makes it useless as a readinessProbe. Consider changing
+		// to only become healthy after the proxy is fully synced.
```

Comment:
@danwinship is this more complicated than just removing this lastUpdated.IsZero() block? lastUpdated will never be updated unless the proxy has fully initialized (eg gotten all initial endpoints and services).

Reply:
in any case, that's a todo...